Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyStringComboBoxCtrl.cpp

```diff
@@ -12,170 +12,71 @@ AZ_PUSH_DISABLE_WARNING(4251, "-Wunknown-warning-option") // 'QLayoutItem::align
-
-        bool indexWasFound = false;
-        if (str.empty())
-        {
-            // Value may just not be populated. Don't warn if that's the case.
-            indexWasFound = true;
-            m_pComboBox->setCurrentIndex(-1);
-        }
-
-        for (size_t enumValIndex = 0; enumValIndex < m_values.size(); enumValIndex++)
-        {
-            if (m_values[enumValIndex].compare(str.c_str()) == 0)
-            {
-                m_pComboBox->setCurrentIndex(static_cast<int>(enumValIndex));
-                indexWasFound = true;
-                break;
-            }
-        }
-
-        if (!indexWasFound)
-        {
-            if (m_pComboBox->isEditable())
-            {
-                m_pComboBox->setEditText(QString(str.c_str()));
-            }
-            else
-            {
-                // item not found void out the combo box
-                m_pComboBox->setCurrentIndex(-1);
-            }
-        }
-
-        AZ_Warning("AzToolsFramework", indexWasFound == true, "No index in property enum for value %s", str.c_str());
-
-        m_pComboBox->blockSignals(false);
-    }
-
-    int PropertyStringComboBoxCtrl::GetCount() const
-    {
-        return m_pComboBox->count();
+        return GetComboBox()->count();
     }
 
-    uint32_t PropertyStringComboBoxCtrl::GetCurrentIndex() const
+    uint32_t PropertyStringComboBoxCtrl::GetCurrentIndex()
     {
-        return static_cast<uint32_t>(m_pComboBox->currentIndex());
+        return static_cast<uint32_t>(GetComboBox()->currentIndex());
     }
 
     void PropertyStringComboBoxCtrl::Add(const AZStd::string& val)
     {
-        m_pComboBox->blockSignals(true);
-        m_values.push_back(val);
         if (val != "---")
         {
-            m_pComboBox->addItem(val.c_str());
+            const AZStd::pair<AZStd::string, AZStd::string> valueToAdd = AZStd::pair(val, val);
+            ComboBoxBase::addElement(valueToAdd);
         }
         else
         {
-            m_pComboBox->insertSeparator(m_pComboBox->count());
+            GetComboBox()->blockSignals(true);
+            GetComboBox()->insertSeparator(GetCount());
+            GetComboBox()->blockSignals(false);
         }
-        m_pComboBox->blockSignals(false);
     }
 
     void PropertyStringComboBoxCtrl::Add(const AZStd::vector<AZStd::string>& vals)
     {
-        m_pComboBox->blockSignals(true);
-        m_values.clear();
-        m_pComboBox->clear();
+        GetComboBox()->clear();
         for (size_t valIndex = 0; valIndex < vals.size(); valIndex++)
         {
-            m_values.push_back(vals[valIndex]);
-            if (vals[valIndex] != "---")
+            auto value = vals[valIndex];
+            if (value != "---")
             {
-                m_pComboBox->addItem(vals[valIndex].c_str());
+                const AZStd::pair<AZStd::string, AZStd::string> valueToAdd = AZStd::pair(value.c_str(), value.c_str());
+                ComboBoxBase::addElement(valueToAdd);
```

Comment:
The `value.c_str()` is a anti-optimization that is unnecessary.
The vals vector is already an `AZStd::string`, no need to grab the `const char*` from it, where it would have to recalculate the string length to create an AZStd::string
```c++
const AZStd::pair<AZStd::string, AZStd::string> valueToAdd{ value, value };
```